### DISCUSSION TITLE
Curious about what broke.

### DISCUSSION THREAD

**Discussion by dbell669** [awawa-dev/HyperSerialPico#11] [Q&A] [answered]
So I was attempting to fix my video solution with a splitter to swap between my gaming pc and my avr. and I noticed that my Pico and attached LED strip was not showing properly. I tried turning it off and on and it appears my pc no longer recogonizes the Pico. I'm wondering if I just broke the Pico or did I fry the strips as well. for context I was using the waveshare RP2040-Zero Mini Board attached to an intel NUC

**Comment by awawa-dev** [comment]
First things first: Pico must be recognized by PC. Did you try to put Pico into DFU mode and reupload the firmware?

**Reply by dbell669** [reply]
Yes. I tried to put it into DFU mode and the NUC couldn't recognize it.

**Reply by awawa-dev** [reply, marked as answer]
Try to connect it (in DFU mode) to another PC. If it's still not recognized in DFU mode then its rp2040 is not working and you must replace the Pico board.

**Reply by dbell669** [reply]
After Attaching it to another computer,. I found that it was fried. I ended up remaking another control module with the larger version, it was said to support a higher current.

**Reply by dbell669** [reply]
I'm now curious as to what fried it in the first place. could it have been a cheap usb cable? or a surge from my PSU for the lights?

**Reply by awawa-dev** [reply]
> it was said to support a higher current.

You should NOT power the LED strip from Pico board! Only the Pico data signal => 3.3V to 5V level shifter (if not built-in) => LED strip data signal. So "support a higher current" doesn't matter anyway.

**Reply by awawa-dev** [reply]
> I'm now curious as to what fried it in the first place. could it have been a cheap usb cable? or a surge from my PSU for the lights?

Maybe it's a random failure that can also happen to Pico, as it happens with esp. Customized Waveshare are not the best, they suffer from similar issues as early Adafruit Pico custom designs, although Adafruit was able to admit to mistakes and announce a fix in subsequent revisions, I have not heard of Waveshare doing anything similar. In systems with LED Pico is usually damaged by 5V. Or from the computer through a faulty USB 5 V to 3.3 V regulator (I have not heard or experienced such a defect myself). Or you shorted the data line to 5V on the LED strip somewhere. I am not sure if the level shifter should prevent such damage, depends on the model, but there is a chance it would protect the system.

**Reply by dbell669** [reply]
OK. I have no direct power going into the Pico board outside of the USB connection, Only into the level shifter as per your other diagrams. It was probably a short of some kind as it happened during a power outage.

**Reply by awawa-dev** [reply]
A power outage may be related, but to be safe, check the resistance between the LED strip data and its 5V using a multimeter with the system off before connecting new Pico.

**Reply by smashwatermelon** [reply]
What reading should you get to tell you that connecting a new pico is safe? 

When using a pico with built in level shifter aside from the usb connection to the Pi the only other connection should be from Pico ground to LED strip ground, correct?

**Reply by awawa-dev** [reply]
The connection is forced by the LED strip, not the LED driver, so it doesn't matter whether you use esp32 or pico: 1 data line for Neopixel strips (ws2812b, sk6812...) or 1 data line and 1 clock line for SPI strips (APA102, sk9812). And the ground shared with the ground of the LED strip. And all this while remembering that the connection should be as short and well-secured as possible. So 2 lines for Neopixel and 3 lines for SPI LEDs including the ground.
If you make a short circuit somewhere further, e.g. by bending the LED strip, it may happen that 5v will go to Pico through the data line and damage it exactly as it would damage ESP32. Therefore, in my opinion, you should not use the connectors that come with the LEDs, because the space between the large tin plugs underneath is about a millimeter (in best case scenario) or a fraction thereof and it is easy to short circuit because the LED tape is very flexible..
![obraz](https://github.com/awawa-dev/HyperSerialPico/assets/69086569/aadb5e21-4cdc-4643-b971-0c62fed25fbe)
